scripts/release/check_sync_criteria.py

```diff
@@ -96,20 +96,28 @@ def check_sync_criteria(
         for pkg_name in sorted(build_file.sync_packages):
             print('-', pkg_name)
 
-    # check that count is satisfied
-    if build_file.sync_package_count is not None:
+    # check that count is satisfied; if it exists, sync_package_percent is
+    # preferred over sync_package_count
+    packages_needed = None
+    if build_file.sync_package_percent is not None:
+        packages_needed = int(len(pkg_names) * build_file.sync_package_percent)
+    if packages_needed is None and build_file.sync_package_count is not None:
+        packages_needed = build_file.sync_package_count
```

Comment:
The `packages` configuration works with `package_percent` by adding an additional criteria for the sync to pass. I think it would be intuitively better to use that same behavior here, and set the threshold to whichever of `package_percent` or `package_count` ends up higher.

Reply:
I struggled with deciding what to do here when you have both `sync_package_count` and `sync_package_percent` specified.  The options as I see them are:

1. If both are specified, set the threshold to the higher of the two calculated values.  This is what you are suggesting above.  The problem I see with this is that it can be harder to debug when a sync doesn't happen.
2. If both are specified, set the threshold to just use `sync_package_percent`.  This is what this PR implements.  The debugging also kind of sucks here, though it is more consistent with how the `packages`/`sync_package_count` currently works.
3. If both are specified, throw an error.  This is the easiest to debug, and is always unambiguous.  But it may be somewhat annoying to the user in that they have to change the configuration to fix it.

I guess the debugging problems in 1 and 2 can be alleviated by adding additional logging, so I'll probably at least go ahead and add that.

All of that said, I actually don't feel strongly about which way we go here.  All 3 options above seem viable to me.  So if we can come to consensus on which way to go, I'm happy to implement it.  @nuclearsandwich any thoughts here?

Reply:
I can buy the debug-ability argument. We can shelve this an move on if you like.

Reply:
After discussing this, we came to the conclusion that all options should be ANDed together.

That is, if all of `packages`, `package_count`, and `package_percent` are specified, the specified packages must be available, AND we must have at least `package_count` packages, AND that count of packages must be at least `package_percent` of the total enabled packages.

Other combinations are of course possible, but they would always be ANDed together.